grafana/grafana
# Elasticsearch: Fix creating of legend so it is backward compatible with frontend produced frames

During backend migration, we focused on making sure that legends generated on frontend and backend are the same. We were able to achieve this trough https://github.com/grafana/grafana/pull/66039, but unfortunately, this solution does not hold when it comes to using transformations - related to https://github.com/grafana/support-escalations/issues/8787. When transformations that rename fields are used, the renaming does not work because `DisplayNameFromDS` overwrites it.

So I was checking if there are other options and I was able to achieve the same frames naming using `frame.Name` and deleting `labels` for `data.FrameTypeTimeSeriesMulti`. 

Here are comparisons on frame created in frontend (left) and backend (right) - both have the same name and no labels:

<img width="1593" alt="image" src="https://github.com/grafana/grafana/assets/30407135/f3a0947c-8655-4d96-915f-543c63fd55a2">

How to test: 
1. Run make devenv sources=elastic
1. Run cd devenv and then ./setup.sh
Go to http://localhost:3000/d/es_complex_template/datasource-tests-elasticsearch-complex-with-template-variables?orgId=1 and make sure that legends on frontend running queries and backend running queries are same (see the recording bellow). 
For now, ignore  **Multiple metrics and aggregations section** which was skipped also in https://github.com/grafana/grafana/pull/66039 and it is not related to this change - it is related to label named `metric`. I will address this in separate PR. 


https://github.com/grafana/grafana/assets/30407135/bf584f0a-7cad-4410-9bba-abba19b79907
